<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>559</width>
    <height>645</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_9">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout" stretch="1,0,0,0,4,1">
      <property name="spacing">
       <number>10</number>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QTextBrowser" name="textBrowser">
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.AppleSystemUIFont'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;By the constraint f(x) &amp;lt; 1, try to propose different distributions on a larger domain and evaluate the performance of this method via the average number of rejected points.(Input the function and sample size with Python syntax.)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>f(x)=</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_function">
          <property name="text">
           <string>(xx**2) * (1 + np.sin(3*yy) * np.cos(8*xx)) + (yy**2) * (2 + np.cos(5*xx) * np.cos(8*yy)) </string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="label_simux">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Simulation of x =</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_simuy">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Simulation of y =</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QComboBox" name="comboBox_xdist">
            <item>
             <property name="text">
              <string>Uniform(-1.5, 1.5)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Uniform(0, 1)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Uniform(-3, 3)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Normal(2, 4)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Normal(0, 1)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Beta(2, 2)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Beta(4, 2)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Beta(2, 4)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Gamma(3, 8)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Gamma(8, 2)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Gamma(6, 6)</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_ydist">
            <item>
             <property name="text">
              <string>Uniform(-1, 1)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Gamma(3, 8)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Gamma(8, 2)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Gamma(6, 6)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Uniform(0, 1)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Normal(-2, 1)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Normal(0, 1)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Normal(2, 5)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Normal(-2, 3)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Beta(1, 1)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Beta(3, 8)</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label_sample">
              <property name="font">
               <font>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Sample size=</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_sample_page3">
              <property name="text">
               <string>10**5</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QSlider" name="hSlider_sample_page3">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="minimum">
             <number>1000</number>
            </property>
            <property name="maximum">
             <number>100000</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="pageStep">
             <number>1000</number>
            </property>
            <property name="value">
             <number>100000</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBothSides</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="2">
        <item>
         <widget class="QPushButton" name="pushButton_enter">
          <property name="text">
           <string>ENTER</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="PlotWidget" name="graphicsView_page3"/>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="2,1">
        <item>
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QTextBrowser" name="textBrowser_result">
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.AppleSystemUIFont'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;The performance of this method via the average number of rejected points is 42.675%&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_exit">
          <property name="text">
           <string>EXIT</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>559</width>
     <height>36</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
